include/cuco/static_map.cuh

```diff
@@ -192,8 +202,19 @@ class static_map {
    * @param stream Stream used for executing the kernels
    */
   static_map(std::size_t capacity,
-             Key empty_key_sentinel,
-             Value empty_value_sentinel,
+             sentinel::empty_key<Key> empty_key_sentinel,
+             sentinel::empty_value<Value> empty_value_sentinel,
+             Allocator const& alloc = Allocator{},
+             cudaStream_t stream    = 0);
+
+  /**
+   * @brief Construct a fixed-size map with erase capability
+   * empty_key_sentinel and erased_key_sentinel must be different values.
```

Comment:
Will this throw if they are they same? We should document that with `@throws` and indicate which exception will be thrown.